### PR TITLE
fix javadoc issues with doclint

### LINKED ISSUE #2872: jHipster fails on Java 8 javadoc compliance
I noticed when performing a maven release that the javadoc written in jHipster is not Java8 compliant.

Some errors encountered : 

- error: self-closing element not allowed
- error: unexpected end tag: </p>
- error: end tag missing: </a>
- error: bad use of '>'

Some options : 

- Fix all javadocs
- Skip javadoc generator in the pom.xml

### PR DESCRIPTION
**Pull request by notflorian**
I added missing @param, @return and @throws,
replaced `->` by `:` in the webservices javadoc to fix the error `bad use of '>'`
replaced `<br/>` and `<p/>` by `<p>` to fix the error `self-closing element not allowed`
fixed `<ul>` and `<li> `tags to fix the errors `unexpected end tag` and `end tag missing`
Finally, I updated the travis scripts to generate the javadoc during the build, so the build should fail if there is error during the generation

Fix #2872

### CODE REVIEW

**Comment by notflorian** [comment]
What do you think of the documentation for the `@return` in the RestControllers?

`     * @return the ResponseEntity with status 200 (OK) and with body the relationshipTestEntity, or with status 404 (Not Found)`
	 
Is it ok to mention the ResponseEntity and the possible HTTP status, or do you prefer I mention only the entity or list of entities?

**Comment by notflorian** [comment]
Looks like the travis buld #3302.9 is randomly failing, because the build is green on the branch on my fork. I will rebase my PR tomorrow to force the travis build to start again.

**Comment by pascalgrimaud** [comment]
You can close / reopen the PR

File: generators/server/templates/src/main/java/package/config/liquibase/_AsyncSpringLiquibase.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 /**
  * Specific liquibase.integration.spring.SpringLiquibase that will update the database asynchronously.
  * <p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/config/liquibase/_AsyncSpringLiquibase.java]: This paragraph tag is never closed

File: generators/server/templates/src/main/java/package/config/liquibase/_AsyncSpringLiquibase.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
...
- *     By default, this asynchronous version only works when using the "dev" profile.<br/>
+ *     By default, this asynchronous version only works when using the "dev" profile.<p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/config/liquibase/_AsyncSpringLiquibase.java]: Neither is this

File: generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java [hunk 3]
```diff
@@ -27,9 +27,9 @@
 
 /**
  * Custom implementation of Spring Security's RememberMeServices.
- * <p/>
+ * <p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java]: Never closed

File: generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java [hunk 4]
```diff
@@ -37,17 +37,16 @@
  * <li>It stores more information, such as the IP address and the user agent, for audit purposes<li>
  * <li>When a user logs out, only his current session is invalidated, and not all of his sessions</li>
  * </ul>
- * <p/>
+ * <p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java]: Same here

File: generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java [hunk 5]
```diff
@@ -137,7 +136,7 @@ protected void onLoginSuccess(HttpServletRequest request, HttpServletResponse re
 
     /**
      * When logout occurs, only invalidate the current token, and not all user sessions.
-     * <p/>
+     * <p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/security/_CustomPersistentRememberMeServices.java]: Same here

File: generators/server/templates/src/main/java/package/web/rest/util/_PaginationUtil.java [hunk 6]
```diff
@@ -10,9 +10,8 @@
  * Utility class for handling pagination.
  *
  * <p>
- * Pagination uses the same principles as the <a href="https://developer.github.com/v3/#pagination">Github API</api>,
+ * Pagination uses the same principles as the <a href="https://developer.github.com/v3/#pagination">Github API</a>,
  * and follow <a href="http://tools.ietf.org/html/rfc5988">RFC 5988 (Link header)</a>.
- * </p>
```
Comment (andidev) [inline generators/server/templates/src/main/java/package/web/rest/util/_PaginationUtil.java]: Paragraph tag should be closed here

**Comment by andidev** [comment]
I did a fast review and it looks like some paragraph tags are not properly closed. 
Shouldn't the Travis tests fail on this?

**Comment by notflorian** [comment]
In fact, the` <p>` elements are here to separate the paragraphs, and doesn't need to be closed.
Here a very interresting explaination I found on [StackOverflow](http://stackoverflow.com/questions/5260368/javadoc-paragraph-separator) :

> Welcome to the land of HTML 3.2.
> 
> According to the official guide on writing doc comments, the correct way to separate paragraphs is with the paragraph tag: `<P>`. Take a look at the seventh bullet in the section on [Format of a Doc Comment](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format).
> 
> Ordinarily, I would strongly recommend against using such old, outdated practices for markup. However, in this case, there's a decent reason to make an exception. The JavaDoc tool (unless radically updated with custom Doclets) generates old, crufty, somewhat broken markup. Browsers have been built to be backwards-compatible with the crazy old markup of the day, so it makes sense for you to just go along with it. Your use of `<P>` to separate paragraphs will be in line with the rest of the JavaDoc output.

**Comment by andidev** [comment]
Sorry you are right! My frontend developer eyes hurts though 😨

**Comment by andidev** [comment]
Let's merge this than!

**Pull request merged**